From: Jason Dillaman <dillaman@redhat.com>
Date: Thu, 19 May 2016 10:11:12 -0400
Subject: journal: reset watch step after pruning expired tag

Signed-off-by: Jason Dillaman <dillaman@redhat.com>
(cherry picked from commit 48c2f4e079000021b6ebe807bb1dba0dc1a6a11d)
(cherry picked from commit 5f09b9aef8f30d9dc7e5ec934b228d885e6456e2)
---
 src/journal/JournalPlayer.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/journal/JournalPlayer.cc b/src/journal/JournalPlayer.cc
index cb728ed..426624c 100644
--- a/src/journal/JournalPlayer.cc
+++ b/src/journal/JournalPlayer.cc
@@ -494,6 +494,7 @@ bool JournalPlayer::prune_active_tag() {
   uint64_t tag_tid = *m_active_tag_tid;
   m_active_tag_tid = boost::none;
   m_splay_offset = 0;
+  m_watch_step = WATCH_STEP_FETCH_CURRENT;
 
   return prune_tag(tag_tid);
 }
